src/components/duration.rs

```diff
@@ -451,6 +452,225 @@ impl Duration {
         }
     }
 
+    #[inline]
+    pub fn round_v2<C: CalendarProtocol, Z: TzProtocol>(
```

Comment:
About the `round_v2`, I think it should just replace the old rounding function. We just need to test `Boa` against the PR branch to see that everything looks good.